File: anagram/anagram.cpp
#include <iostream>
#include <string>

using std::cout;
using std::endl;
using std::string;

bool anagram(string str1, string str2) {
  if (str1.length() != str2.length()) {
    false;
  }
  int char_set[256] = {0};
  for (int i = 0; i < str1.length(); i++) {
    char_set[str1[i]]++;
  }
  for (int i = 0; i < str2.length(); i++) {
    char_set[str2[i]]--;
  }
  for (int i = 0; i < 256; i++) {
    if (char_set[i] != 0) {
      return false;
    }
  }
  return true;
}


int main() {
  string a = "Hello";
  string b = "olelH";
  cout << a << " is ";
  if (!anagram(a, b)) {
    cout << "not";
  }
  cout << " an anagram of " << b << endl;
}

File: anagram/anagram.py
#!/usr/bin/python

from collections import defaultdict


def anagram(a, b):
    if (len(a) != len(b)):
        return False
    char_set = defaultdict(int)
    for char_a, char_b in zip(a, b):
        char_set[char_a] += 1
        char_set[char_b] -= 1
    for char, amount in char_set.iteritems():
        if amount != 0:
            return False
    return True

a = "Hello"
b = "olelH"
print a, "is",
if not anagram(a, b):
    print "not",
print "an anagram of", b
